internal/providers/terraform/azure/util.go

```diff
@@ -63,6 +170,24 @@ func convertRegion(region string) string {
 	}
 }
 
+// GetResourceRegion returns the region for the given azure resource data. By default,
+// this uses the "location" property, but if that is not set it will traverse the
+// references to find a location. The default references used are the resource
+// group name. However, some resources specify different resources to get the
+// location from, in which case a custom function is used that specifies which
+// references to use.
+func GetResourceRegion(d *schema.ResourceData) string {
```

Comment:
2 other options here:

1. Should we change the `terraform/azure/*.go` resources to call this function instead of `lookupRegion` directly?

2. Should we put the `lookupRegion` function in the `RegistryItem` struct like we do for `ReferenceAttributes`? That way it can live alongside the other Terraform resource code in the `terraform/azure/*.go`

Reply:
1. technically we don't have to now, we can just use the `Region` property on the `SchemaData` which is now set. This will be correct for each resource. I initially left this out because I thought it might be too big a change, but happy to change all the areas we use `lookupRegion` to use the property instead
2. Yeah I guess that would maybe work better

Reply:
I have now changed this function so that it only gets the default fallback for the provider (azure). In `terraform/parser.go` we now call the `RegistryItem.GetRegion` func if it exists otherwise we fallback to this function.

```go
regionFunc := ResourceRegistryMap.GetRegion(d.Type)
if regionFunc != nil {
    return regionFunc(d)
}

 ...
 
 return azure.GetResourceRegion(d)
```

Reply:
Additionally all azure resources now use the `d.Region` property at runtime rather than the `lookupRegion`.